File: [Manya]/Leetcode/Sliding-Window/1358.java
import java.util.Scanner;

class Solution {
    public int numberOfSubstrings(String s) {
        int count[] = {0, 0, 0};
        int left = 0;
        int result = 0;

        for (int right = 0; right < s.length(); right++) {
            count[s.charAt(right) - 'a']++;

            while (count[0] > 0 && count[1] > 0 && count[2] > 0) {
                result += s.length() - right;
                count[s.charAt(left) - 'a']--;
                left++;
            }
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter the string: ");
        String s = scanner.next();

        Solution solution = new Solution();
        int result = solution.numberOfSubstrings(s);
        
        System.out.println("Number of substrings containing 'a', 'b', and 'c': " + result);

        scanner.close();
    }
}

File: Aditya-Pratap/Sliding Window/1358.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        A,B,C=-1,-1,-1
        count=0
        for i in range(len(s)):
            if s[i]=="a":
                A=i
            if s[i]=="b":
                B=i
            if s[i]=="c":
                C=i
            if A!=-1 and B!=-1 and C!=-1:
                count+=1+min(A,B,C)
        return count